tests/unittests/CoreGraphics.drawing/CGContextTransparencyLayerTests.cpp

```diff
@@ -0,0 +1,183 @@
+    CGContextAddLineToPoint(context, center.x, center.y + 50);
+    CGContextStrokePath(context);
+
+    CGContextEndTransparencyLayer(context);
+}
+
+DRAW_TEST_F(CGContext, TransparencyLayerWithInnerShadowCanStack, WhiteBackgroundTest) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGContextSetShadow(context, CGSize{ 10.f, 10.f }, 1.0);
+
+    CGContextBeginTransparencyLayer(context, nullptr);
+
+    CGContextSetShadow(context, CGSize{ -10.f, -10.f }, 1.0);
+
+    CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextSetLineWidth(context, 5);
+
+    _strokeTwoCirclesInContext(context, bounds);
+
+    CGContextEndTransparencyLayer(context);
+}
+
+DRAW_TEST_F(CGContext, TransparencyLayerIsAlphaCompositedAllAtOnce, WhiteBackgroundTest) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGContextSetAlpha(context, 0.5);
+
+    CGContextBeginTransparencyLayer(context, nullptr);
+
+    CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextSetLineWidth(context, 5);
+
+    _strokeTwoCirclesInContext(context, bounds);
+
+    CGContextEndTransparencyLayer(context);
+}
+
+DRAW_TEST_F(CGContext, TransparencyLayerWithZeroGlobalAlpha, WhiteBackgroundTest) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGContextSetAlpha(context, 0.0);
+
+    CGContextBeginTransparencyLayer(context, nullptr);
+
+    CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextSetLineWidth(context, 5);
+
+    _strokeTwoCirclesInContext(context, bounds);
+
+    CGContextEndTransparencyLayer(context);
+}
+
+DRAW_TEST_F(CGContext, TransparencyLayerWithRect, WhiteBackgroundTest) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGPoint center = _CGRectGetCenter(bounds);
+    CGRect clippingRect = _CGRectCenteredOnPoint({ 150, 150 }, center);
+
+    CGContextBeginTransparencyLayerWithRect(context, clippingRect, nullptr);
+
+    CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextSetLineWidth(context, 5);
+
+    _strokeTwoCirclesInContext(context, bounds);
+
+    CGContextEndTransparencyLayer(context);
+}
+
+// The shadow should extend beyond the bounds of the layer content, even if it is clipped.
+DRAW_TEST_F(CGContext, TransparencyLayerWithRectAndShadow, WhiteBackgroundTest) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGPoint center = _CGRectGetCenter(bounds);
+    CGRect clippingRect = _CGRectCenteredOnPoint({ 100, 100 }, center);
+
+    CGContextSetShadow(context, CGSize{ 10.f, 10.f }, 1.0);
+
+    CGContextBeginTransparencyLayerWithRect(context, clippingRect, nullptr);
+
+    CGContextSetRGBFillColor(context, 1.0, 0.0, 0.0, 1.0);
+    CGContextFillRect(context, bounds);
+    //CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
```

Comment:
good catch.

---
In reply to: [90587281](https://github.com/Microsoft/WinObjC/pull/1496#discussion_r90587281) [](ancestors = 90587281)